htdocs/includes/odtphp/odf.php

```diff
@@ -18,980 +18,1003 @@ class OdfException extends Exception
+            //unoconv: RuntimeException during import phase:
+            //Office probably died. Unsupported URL <file:///tmp/document-example.odt>: "type detection failed"
+            //DEBUG: Terminating LibreOffice instance.
+            //DEBUG: Waiting for LibreOffice instance to exit
+
+            // If it fails:
+            // - set shell of user to bash instead of nologin.
+            // - set permission to read/write to user on home directory /var/www so user can create the libreoffice , dconf and .cache dir and files then set permission back
+
+            $command = getDolGlobalString('MAIN_ODT_AS_PDF').' '.escapeshellcmd($name);
+            //$command = '/usr/bin/unoconv -vvv '.escapeshellcmd($name);
+        } else {
+            // deprecated old method using odt2pdf.sh (native, jodconverter, ...)
+            $tmpname=preg_replace('/\.odt/i', '', $name);
+
+            if (getDolGlobalString('MAIN_DOL_SCRIPTS_ROOT')) {
+                $command = getDolGlobalString('MAIN_DOL_SCRIPTS_ROOT').'/scripts/odt2pdf/odt2pdf.sh '.escapeshellcmd($tmpname).' '.(is_numeric(getDolGlobalString('MAIN_ODT_AS_PDF'))?'jodconverter':getDolGlobalString('MAIN_ODT_AS_PDF'));
+            } else {
+                dol_syslog(get_class($this).'::exportAsAttachedPDF is used but the constant MAIN_DOL_SCRIPTS_ROOT with path to script directory was not defined.', LOG_WARNING);
+                $command = '../../scripts/odt2pdf/odt2pdf.sh '.escapeshellcmd($tmpname).' '.(is_numeric(getDolGlobalString('MAIN_ODT_AS_PDF'))?'jodconverter':getDolGlobalString('MAIN_ODT_AS_PDF'));
+            }
+        }
+
+        //$dirname=dirname($name);
+        //$command = DOL_DOCUMENT_ROOT.'/includes/odtphp/odt2pdf.sh '.$name.' '.$dirname;
+
+        dol_syslog(get_class($this).'::exportAsAttachedPDF $execmethod='.$execmethod.' Run command='.$command, LOG_DEBUG);
+        // TODO Use:
+        // $outputfile = DOL_DATA_ROOT.'/odt2pdf.log';
+        // $result = $utils->executeCLI($command, $outputfile);  and replace test on $execmethod.
+        // $retval will be $result['result']
+        // $errorstring will be $result['output']
+        $retval=0; $output_arr=array();
+        if ($execmethod == 1) {
+            exec($command, $output_arr, $retval);
+        }
+        if ($execmethod == 2) {
+            $outputfile = DOL_DATA_ROOT.'/odt2pdf.log';
+
+            $ok=0;
+            $handle = fopen($outputfile, 'w');
+            if ($handle) {
+                dol_syslog(get_class($this)."Run command ".$command, LOG_DEBUG);
+                fwrite($handle, $command."\n");
+                $handlein = popen($command, 'r');
+                while (!feof($handlein)) {
+                    $read = fgets($handlein);
+                    fwrite($handle, $read);
+                    $output_arr[]=$read;
+                }
+                pclose($handlein);
+                fclose($handle);
+            }
+            dolChmod($outputfile);
```

Comment:
With v17, this line of code will hang. Keep v17 code as this has no relation with the bugfix.

Reply:
Done